hq/app/logic/IamRemediation.scala

```diff
@@ -45,19 +45,23 @@ object IamRemediation extends Logging {
       credentialReportDisplay.humanUsers.filter(user => hasOutdatedHumanKey(List(user.key1, user.key2), now))
   }
 
-  private def hasOutdatedHumanKey(keys: List[AccessKey], now: DateTime): Boolean = keys.exists { key =>
-      key.lastRotated.exists { date =>
-        // using minus 1 so that we return true if the last rotated date is exactly on the cadence date
-        date.isBefore(now.minusDays(Config.iamHumanUserRotationCadence.toInt - 1))
-      } && key.keyStatus == AccessKeyEnabled
-    }
+  private def hasOutdatedHumanKey(keys: List[AccessKey], now: DateTime): Boolean = keys.exists(isOutdatedHumanKey(_, now))
 
-  private def hasOutdatedMachineKey(keys: List[AccessKey], now: DateTime): Boolean = keys.exists { key =>
-      key.lastRotated.exists { date =>
-        // using minus 1 so that we return true if the last rotated date is exactly on the cadence date
-        date.isBefore(now.minusDays(Config.iamMachineUserRotationCadence.toInt - 1))
-      } && key.keyStatus == AccessKeyEnabled
-    }
+  private def hasOutdatedMachineKey(keys: List[AccessKey], now: DateTime): Boolean = keys.exists(isOutdatedMachineKey(_, now))
+
+  private def isOutdatedHumanKey(key: AccessKey, now: DateTime): Boolean = {
+    key.lastRotated.exists { date =>
+      // using minus 1 so that we return true if the last rotated date is exactly on the cadence date
```

Comment:
These datetimes have second resolution, so this might not quite behave as expected. It's not a big deal though, if everything happens one day earlier or later than expected for a specific key it's still fine. Let's consider if we want to be more precise later on, we're absolutely fine here for now!